lib/ll/setup.rb

```diff
@@ -3,7 +3,16 @@
 require 'll/driver_config'
 require 'll/parser_error'
 require 'll/configuration_compiler'
-require 'libll'
+
+if RUBY_PLATFORM == 'opal'
+  require 'll/native/driver'
+  require 'll/native/driver_config'
+elsif ENV['RUBYLL_PURERUBY']
```

Comment:
>Is ext/pureruby an Opal thing?

No, it is not. I added this path for logical equivalence with other extension implementation. We can move it to `lib` if that's in your opinion a better location for it.

>To add to that: it's best to merge this `if` with the one for Java below

As of currently, the particular sexp of `if RUBY_PLATFORM == "opal"` is a special macro that gates off the code from compilation on Opal side. In particular, the `requires` that are in this part are not compiled into the resulting `.js` bundle. This can, of course, be extended to `case`.

But what I in particular wanted to achieve here with the environment variable is to be able to test this library on MRI.